.github/workflows/prune.yml

```diff
@@ -10,19 +10,74 @@ on:
     - cron: "0 0 * * *"  # every day at midnight
 
 jobs:
-  build:
+  # Phase 1: find all docker image names and release tags
+  fetch_image_info:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Fetch all aiSSEMBLE docker image names and release versions, then save to respective outputs
+        id: fetch-image-info
+        env:
+          GH_TOKEN: ${{ secrets.GHCR_IO_TOKEN }}
 
+        # Find all docker image names that starts with "aissemble" and does not end with "-chart" and save them to matrix-values output
+        # And find all release version tags and saves them to release-values output
+        run: |
+          image_names=$(gh api \
+            -H "Accept: application/vnd.github+json" \
+            -H "-GitHub-Api-Version: 2022-11-28" \
+            -H "Authorization: Token ${GH_TOKEN}" \
+            --paginate "/orgs/boozallen/packages?package_type=container" | \
+            jq -c '[.[] | select((.name | startswith("aissemble")) and (.name | endswith("-chart") | not)) | .name]')
+          
+          # save to output
+          echo "matrix_values=${image_names}" >> $GITHUB_OUTPUT
+  
+          # loop through each image name and find all release tags using regex pattern
+          for name in $(echo "$image_names" | jq -r '.[]'); do
+              release_versions=$(gh api \
+                -H "Accept: application/vnd.github+json" \
+                -H "-GitHub-Api-Version: 2022-11-28" \
+                -H "Authorization: Token ${GH_TOKEN}" \
+                --paginate "/orgs/boozallen/packages/container/${name}/versions" \
+                | jq -r '.[] | .metadata.container.tags[] | select(test("^\\d+\\.\\d+\\.\\d+$"))')
+          
+              # append all release tags to all_release_versions array
+              all_release_versions+=("${release_versions}")
+          done
+          
+          # replace all whitespaces with newline (required for sort) and then sort all release versions while removing any duplicates
+          all_release_versions_sorted=$(echo "${all_release_versions[@]}" | tr ' ' '\n'| sort -u )
+          
+          # replace the newlines with commas and then remove the trailing comma
+          all_release_versions=$(echo "${all_release_versions_sorted[@]}" | tr '\n' ',' | sed 's/,$//')
```

Comment:
I: At this point, `all_release_versions` looks something like this:
```
1.7.0
1.8.0
```
To achieve the format we need, we replace all `\n` with commas to get:
```
1.7.0,1.8.0,
```
Lastly, we need to remove the trailing comma to get:
```
1.7.0,1.8.0
```